controllers/idler/idler_controller.go

```diff
@@ -150,13 +156,15 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 					appName = pod.Name
 					appType = "Pod"
 				}
-
-				// By now either a pod has been deleted or scaled to zero by controller, idler Triggered notification should be sent
-				if err := r.createNotification(logger, idler, appName, appType); err != nil {
-					logger.Error(err, "failed to create Notification")
-					if err = r.setStatusIdlerNotificationCreationFailed(idler, err.Error()); err != nil {
-						logger.Error(err, "failed to set status IdlerNotificationCreationFailed")
-					} // not returning error to continue tracking remaining pods
+				// Do not send notification if Pod Not managed by a controller and is in completed state
+				if podreason != "PodCompleted" && !deletedByController {
```

Comment:
Shouldn't it be
```suggestion
				if podreason != "PodCompleted" || deletedByController {
```
instead?
Your current statement will result in creating a notification only in the case of standalone pods in a non-complete state. We still want to create notifications for pods scaled down to zero through their controllers.

And please also make sure that all those use cases are covered by the unit tests:
- Standalone pod with "PodCompleted" reason.
- Standalone pod with a non-complete reason.
- Pod managed by a controller with "PodCompleted" reason.
- Pod managed by a controller with a non-complete reason.

Reply:
we want to send notification for individual pods which are without controllers and are non-complete state ..? If yes than it cannot be || condition, else for any pod without controller there will be no notification

Reply:
> we want to send notification for individual pods which are without controllers and are non-complete state ..?

Yes. But we also want to send a notification for pods owned by a controller. No matter what state.

> If yes than it cannot be || condition, else for any pod without controller there will be no notification.

Not really. There will be no notification only if both conditions are met: pod is complete AND not owned by a controller (the "if" will be false in that case, so the notification is skipped).

Reply:
will do